source/framework/vx_graph_verify.c

```diff
@@ -730,98 +727,148 @@ static vx_status ownGraphCalcInAndOutNodes(vx_graph graph)
+                                        VX_PRINT(VX_ZONE_ERROR,"Output of node connected to same node at index %d failed\n", node_cur_idx);
+                                    }
+                                }
+                                else if (prm_cur_dir == prm_next_dir)
+                                {
+                                    /* We have an error: two modifiers or two writers of the same edge */
+                                    status = (vx_status)VX_FAILURE;
+                                    VX_PRINT(VX_ZONE_ERROR,"Two modifiers or two writers of the same edge, nodes %d and %d\n", node_cur_idx, node_next_idx);
+                                }
+                                else if ((uint32_t)VX_INPUT == prm_next_dir)
+                                {
+                                    ++inputs_attached;
                                     /* add node_next as output node for current node if not already added */
-                                    status = ownNodeAddOutNode(node_cur, node_next);
-
-                                    if(status == (vx_status)VX_SUCCESS)
+                                    /* but we don't do it if this parameter is an output attached to a bidirectional */
+                                    if (0 == biputs_attached)
                                     {
-                                        /* add node_current as input node for next node if not already added */
-                                        status = ownNodeAddInNode(node_next, node_cur);
-                                        if (status != (vx_status)VX_SUCCESS)
+                                        status = ownNodeAddOutNode(node_cur, node_next);
+                                        if(status == (vx_status)VX_SUCCESS)
                                         {
-                                            VX_PRINT(VX_ZONE_ERROR,"Add in node at index %d failed\n", node_cur_idx);
+                                                /* add node_current as input node for next node if not already added */
+                                                status = ownNodeAddInNode(node_next, node_cur);
+                                                if (status != (vx_status)VX_SUCCESS)
+                                                {
+                                                    VX_PRINT(VX_ZONE_ERROR,"Add in node at index %d failed\n", node_cur_idx);
+                                                }
+                                        }
+                                        else
+                                        {
+                                            VX_PRINT(VX_ZONE_ERROR,"Add out node at index %d failed\n", node_cur_idx);
                                         }
                                     }
+                                }
+                                else if (VX_BIDIRECTIONAL == prm_next_dir)
```

Comment:
It looks like there are some casts missing still

Reply:
add the cast @RB